### PR TITLE
KafkaService updated.

### PR DESCRIPTION
**Pull request by ByBordex**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/Arquisoft/InciManager_i3a/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Now a service is available for sending messages to apache Kafka
Issue Number: #7 

## Does this PR introduce a breaking change?
```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/org/uniovi/i3a/incimanager/kafka/KafkaProduceFactory.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+	@Bean
+	public ProducerFactory<String, String> producerFactory() {
+		return new DefaultKafkaProducerFactory<>(producerConfigs());
+	}
+
+	@Bean
+	public Map<String, Object> producerConfigs() {
+		Map<String, Object> props = new HashMap<>();
+		props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
```
Comment (thewilly) [inline src/main/java/org/uniovi/i3a/incimanager/kafka/KafkaProduceFactory.java]: Would be awesome if the address and the port of kafka where read from the properties file. In that case you can specify a local resource for testing and a remote resource for CD. You have one example in the rest services.

File: src/main/java/org/uniovi/i3a/incimanager/kafka/KafkaService.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import org.springframework.kafka.support.SendResult;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+@ManagedBean
+@EnableAsync
+public class KafkaService {
+
+	private final String TOPIC = "INCIDENCES";
```
Comment (thewilly) [inline src/main/java/org/uniovi/i3a/incimanager/kafka/KafkaService.java]: Sale with the topic, if could came from the proprieties file much better.

File: src/main/java/org/uniovi/i3a/incimanager/kafka/KafkaService.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+@ManagedBean
+@EnableAsync
+public class KafkaService {
```
Comment (thewilly) [inline src/main/java/org/uniovi/i3a/incimanager/kafka/KafkaService.java]: This class should implement the public interface that other services will use

File: src/main/java/org/uniovi/i3a/incimanager/kafka/KafkaService.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+		map.put("asignee", asignee);
+		map.put("state", state);
+		map.put("expiration", expiration);
+		map.put("tags", tags);
+		map.put("additional_information", additionalInfo);
+		map.put("properties", properties);
+		map.put("location", location);
+
+		boolean success = send(TOPIC, new JSONObject(map).toString());
+		return success;
```
Comment (thewilly) [inline src/main/java/org/uniovi/i3a/incimanager/kafka/KafkaService.java]: You could just return send(....) isn’t it? Jejejeje

**Comment by thewilly** [comment]
@ByBordex kafka its not properly configured and topic "INCIDENCES" is not yet configured. Given that build and integration tests should not pass. Even though they pass perfectly. Could you check the tests and then update them?

File: src/main/resources/application-test.properties [hunk 5]
```diff
@@ -1,2 +1,5 @@
 server.port=8090
-agents.url=http://localhost:8080/user
\ No newline at end of file
+agents.url=http://localhost:8080/user
+
+kafka.topic=INCIDENCES
+spring.kafka.producer.bootstrap-servers=localhost:9092
```
Comment (thewilly) [inline src/main/resources/application-test.properties]: Just a doubt, kafka default port is `2181` why change it to 9092?

Comment (thewilly) [reply src/main/resources/application-test.properties]: @ByBordex can you explain if its compulsory to use 9092? just for the ci to work properly.

Comment (ByBordex) [reply src/main/resources/application-test.properties]: I've just followed the startup guide about [sending messages](https://kafka.apache.org/quickstart#quickstart_send) which uses such port.

Comment (thewilly) [reply src/main/resources/application-test.properties]: okey, i guess it works haha

Comment (ByBordex) [reply src/main/resources/application-test.properties]: @thewilly I've made some research and it seems that 2181 is the port of zookeper. So kafka relies in zookeper port but listens to 9092 by defaut.

**Pull request merged**